keps/sig-scheduling/4815-dra-partitionable-devices/README.md

```diff
@@ -252,222 +252,470 @@ non-goals of this KEP.
+  // Name defines the name of the capacity pool.
+  // It must be a DNS label.
+  //
+  // +required
+  Name string
+
+  // Includes defines the set of capacity pool mixins that this capacity
+  // pool includes.
+  //
+  // The propertes of each included mixin are applied to this capacity pool in
+  // order. Conflicting properties from multiple mixins are taken from the
+  // last mixin listed that contains them. Properties set on the capacity pool will
+  // always override properties from mixins.
+  //
+  // The mixins referenced here must be defined in the same
+  // ResourceSlice.
+  //
+  // The maximum number of mixins that can be included is 8.
+  //
+  // +optional
+  // +listType=atomic
+  Includes []CapacityPoolMixinRef
+
+  // Capacity defines the set of capacities for this capacity pool
+  // The name of each capacity must be unique in that set.
+  //
+  // To ensure this uniqueness, capacities defined by the vendor
+  // must be listed without the driver name as domain prefix in
+  // their name. All others must be listed with their domain prefix.
+  //
+  // Capacities listed here will always take precedence over any included
+  // from a mixin.
+  //
+  // The maximum number of capacities is 32.
+  //
+  // +required
+  Capacity map[QualifiedName]DeviceCapacity
+}
+
+// DeviceMixinRef defines a reference to a device mixin. It
+// also lets users provide a capacity pool which will be used
+// to resolve any entries in the SharedCapacityConsumed list
+// in the mixin to the correct capacity pool.
 type DeviceMixinRef struct {
-	// Name refers to the name of a device mixin in the pool.
-	//
-	// +required
-	Name string `json:"name"`
+  // Name refers to the name of a device mixin in the pool.
+  //
+  // +required
+  Name string
+
+  // CapacityPool defines the capacity pool that will be used
+  // when resolving entries in the SharedCapacityConsumed list
+  // in the mixin.
+  //
+  // The value is only needed if the referenced mixin has
+  // at least one entry in the SharedCapacityConsumed list.
+  //
+  // +optional
+  CapacityPool string
+}
+
+// CapacityPoolMixinRef defines a reference from a capacity pool
+// to a capacity pool mixin.
+type CapacityPoolMixinRef struct {
+  // Name is the name of a CapacityPoolMixin.
+  //
+  // +required
+  Name string
+}
+
+// MixinCapacityConsumedRef defines a reference from a mixin
+// to a capacity in an unspecified pool. This allows the same
+// mixin to be used across multiple pools since the pool is
+// defined when the mixin is referenced from the device.
+type MixinCapacityConsumedRef struct {
+  // Name defines the name of the shared capacity that
+  // is being referenced.
+  //
+  // Name must be unique for the references from a single
+  // device, but multiple devices can reference the same
+  // shared capacity.
+  //
+  // +required
+  Name string
+
+  // Capacity defines the capacity that is required by
+  // the device.
+  //
+  // +required
+  Capacity resource.Quantity
```

Comment:
Should this be a DeviceCapacity (to match what is held in an actual CapacityPool) and not just a resource.Quantity?

Reply:
Yeah, I missed this.